src/RequestForm.js

```diff
@@ -318,55 +312,61 @@ class RequestForm extends React.Component {
         this.props.onChangePatron(selectedUser);
         this.setState({ selectedUser });
         this.props.change('requesterId', selectedUser.id);
-        this.findRequestPreferences(selectedUser.id).then(this.setDefaultRequestPreferences);
+        this.findRequestPreferences(selectedUser.id);
       }
     }).then(_ => this.props.asyncValidate());
   }
 
-  findRequestPreferences(userId) {
-    return this.props.findResource('requestPreferences', userId, 'userId')
-      .then((preferences) => {
-        const preference = preferences.requestPreferences[0] || {};
-
-        this.setState({
-          hasDelivery: preference.delivery || false,
-          fulfillmentPreference: preference.fulfillment || fulfilmentTypeMap.HOLD_SHELF,
-          defaultServicePointId: preference.defaultServicePointId || '',
-          defaultDeliveryAddressTypeId: preference.defaultDeliveryAddressTypeId || '',
-          requestPreferencesLoaded: true,
-        });
-      })
-      .catch(() => {
-        this.setState({ requestPreferencesLoaded: true });
-      });
+  getDefaultRequestPreferences() {
+    return {
+      hasDelivery: false,
+      requestPreferencesLoaded: false,
+      defaultDeliveryAddressTypeId: '',
+      defaultServicePointId: '',
+      selectedDelivery: isDelivery(this.props.initialValues),
+      selectedAddressTypeId: this.props.initialValues.deliveryAddressTypeId || '',
+      fulfillmentPreference: fulfilmentTypeMap.HOLD_SHELF,
+    };
   }
 
-  setDefaultRequestPreferences = () => {
-    const {
-      hasDelivery,
-      fulfillmentPreference,
-      defaultServicePointId,
-      defaultDeliveryAddressTypeId,
-    } = this.state;
-
-    this.props.change('fulfilmentPreference', fulfillmentPreference);
-
-    if (fulfillmentPreference === fulfilmentTypeMap.DELIVERY) {
+  async findRequestPreferences(userId) {
+    try {
+      const {
+        requestPreferences: [{
+          delivery: hasDelivery,
+          defaultServicePointId,
+          defaultDeliveryAddressTypeId,
+          fulfillment: fulfillmentPreference,
+        }]
+      } = await this.props.findResource('requestPreferences', userId, 'userId');
+
+      const requestPreferences = {
+        ...this.getDefaultRequestPreferences(),
+        requestPreferencesLoaded: true,
+        hasDelivery,
+        defaultServicePointId,
+        defaultDeliveryAddressTypeId,
+      };
+
+      this.setState({ ...requestPreferences }, () => {
+        this.props.change('fulfilmentPreference', fulfillmentPreference);
+
+        if (isDelivery(this.props.initialValues)) {
```

Comment:
I guess it should be `if (fulfillmentPreference === fulfilmentTypeMap.DELIVERY)`